sky/optimizer.py

```diff
@@ -195,46 +311,40 @@ def _remove_dummy_source_sink_nodes(dag: 'dag_lib.Dag'):
 
     @staticmethod
     def _get_egress_info(
-        parent: task_lib.Task,
         parent_resources: resources_lib.Resources,
-        node: task_lib.Task,
-        resources: resources_lib.Resources,
+        resources: resources_lib.Resources, edge_data: Dict[str, Any]
     ) -> Tuple[Optional[clouds.Cloud], Optional[clouds.Cloud], Optional[float]]:
-        if isinstance(parent_resources.cloud, DummyCloud):
-            # Special case.  The current 'node' is a real
-            # source node, and its input may be on a different
-            # cloud from 'resources'.
-            if node.get_inputs() is None:
-                # A task_lib.Task may have no inputs specified.
-                return None, None, 0
-            src_cloud = node.get_inputs_cloud()
-            nbytes = node.get_estimated_inputs_size_gigabytes()
-        else:
-            src_cloud = parent_resources.cloud
-            nbytes = parent.get_estimated_outputs_size_gigabytes()
+        # TODO(wenjie): Add dummy nodes for external storage when DAG nodes read
+        # from buckets specified in the YAML file.
+        src_cloud = parent_resources.cloud
+        assert isinstance(edge_data['edge'], TaskEdge)
+        task_edge = edge_data['edge']
```

Comment:
can we pass in `edge_data['edge']` instead of `edge_data`? iiuc this is the only entry used in this function.